addons/mail/static/src/models/composer_view.js

```diff
@@ -157,13 +157,11 @@ registerModel({
             // Specific cases for channel and partner mentions: the message with
             // the mention will appear in the target channel, or be notified to
             // the target partner.
-            switch (this.activeSuggestedRecord.constructor.name) {
-                case 'Thread':
-                    Object.assign(updateData, { mentionedChannels: link(this.activeSuggestedRecord) });
-                    break;
-                case 'Partner':
-                    Object.assign(updateData, { mentionedPartners: link(this.activeSuggestedRecord) });
-                    break;
+            if (this.activeSuggestion.thread) {
+                Object.assign(updateData, { mentionedChannels: link(this.activeSuggestion.thread) });
+            }
+            if (this.activeSuggestion.partner) {
+                Object.assign(updateData, { mentionedPartners: link(this.activeSuggestion.partner) });
```

Comment:
I guess both conditions will never be true at the same time because of the identifying fields restriction, but shouldn't that still be made explicit in the code using an `else if`?

Reply:
I prefer to avoid `else (if)` when possible, because this makes following statement(s) have too much context.
This code is too algorithmic. If it was declarative enough, the exclusion should be very clear on just model and field declaration.

Reply:
The exclusion doesn't need to be explicit, it doesn't even need to be taken into account at all.

If both cases ever become true at the same time for any reason, we would be happy that the code already works with both of them.

And while it's not the case, it doesn't hurt (it's actually easier to read simply conditions rather than a chain).

Reply:
in the end I don't really have a strong opinion on this, do as you please 👍️

Reply:
kept version without `else`, just because smaller isolated pieces, and if there's a bug it's better to have too much than too little (and need to debug why we have to little).